apiserver/facades/client/machinemanager/instance_information.go

```diff
@@ -5,69 +5,56 @@ package machinemanager
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/juju/errors"
 
-	"github.com/juju/juju/apiserver/common"
 	apiservererrors "github.com/juju/juju/apiserver/errors"
 	"github.com/juju/juju/core/constraints"
+	"github.com/juju/juju/core/providertracker"
 	"github.com/juju/juju/environs"
-	environscloudspec "github.com/juju/juju/environs/cloudspec"
 	"github.com/juju/juju/environs/envcontext"
 	"github.com/juju/juju/rpc/params"
-	"github.com/juju/juju/state/stateenvirons"
 )
 
 // TODO(wallyworld) - this method is unused by juju; do we still need it?
 
 // InstanceTypes returns instance type information for the cloud and region
 // in which the current model is deployed.
 func (mm *MachineManagerAPI) InstanceTypes(ctx context.Context, cons params.ModelInstanceTypesConstraints) (params.InstanceTypesResults, error) {
-	return instanceTypes(ctx, mm, environs.GetEnviron, cons)
+	invalidatorFunc, err := mm.credentialInvalidatorGetter()
+	if err != nil {
+		return params.InstanceTypesResults{}, errors.Trace(err)
+	}
+	callCtx := envcontext.WithCredentialInvalidator(ctx, invalidatorFunc)
+	return instanceTypes(callCtx, mm.instanceTypeProvider, cons)
 }
 
-type environGetFunc func(context.Context, environs.EnvironConfigGetter, environs.NewEnvironFunc) (environs.Environ, error)
-
+// instanceTypes reports back the results from the provider for what instance
+// types are available for given constraints.
 func instanceTypes(
-	ctx context.Context,
-	mm *MachineManagerAPI,
-	getEnviron environGetFunc,
+	ctx envcontext.ProviderCallContext,
+	providerGetter providertracker.ProviderGetter[environs.InstanceTypesFetcher],
```

Comment:
Replace `providertracker.ProviderGetter[environs.InstanceTypesFetcher]` with the function type `func(context.Context) (environs.InstanceTypesFetcher, error)`, then we can remove the `providertracker` import.

Reply:
The provide tracker is a core package. Why are we adverse to having a core package import? Why do you want to ditch a strong type in favour of this?

Your request needs more justification.